### PR TITLE
Concat all CSS into a single file, and load early.

### PR DESCRIPTION
**Pull request by aidvu**
- Concat all CSS into a single file
- Load the file on top
- Add debugging 
- Add css exclusion list option

### CODE REVIEW

File: concat-css.php [hunk 1]
```diff
@@ -231,6 +235,41 @@ function css_concat_init() {
 
 	$wp_styles = new Page_Optimize_CSS_Concat( $wp_styles );
 	$wp_styles->allow_gzip_compression = ALLOW_GZIP_COMPRESSION;
+
+	// Add a `move-styles` query param to print the styles after the title element
```
Comment (brandonpayton) [inline concat-css.php]: Adding query params so we can test printing styles earlier along with this PR. We can remove the commit later if desired.

File: settings.php [hunk 2]
```diff
@@ -72,6 +72,12 @@ function page_optimize_settings_field_css( $args ) {
 			<input type="checkbox" id="page_optimize-css" name="page_optimize-css" value="1" <?php checked( get_option( 'page_optimize-css' ) ); ?>>
 			<?php _e( 'Concatenate styles', page_optimize_get_text_domain() ); ?>
 		</label>
+		<br />
```
Comment (brandonpayton) [inline settings.php:75]: In regular HTML, we don't need to self-close `<br>` tags. No need to change. Just a comment.
https://html.spec.whatwg.org/#the-br-element

We should also be using CSS instead of `<br>` to separate elements on different lines, but that's a small thing that can be fixed later. I think I may have been the first to use `<br>` here btw... just sharing my thoughts.

File: concat-js.php [hunk 3]
```diff
@@ -146,6 +146,9 @@ function do_items( $handles = false, $group = false ) {
 			foreach ( $exclude_list as $exclude ) {
 				if ( $do_concat && false !== strpos( $handle, $exclude ) ) {
 					$do_concat = false;
+					if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+						echo sprintf( "\n<!-- No Concat JS %s => Excluded option -->\n", esc_html( $handle ) );
```
Comment (brandonpayton) [inline concat-js.php:150]: <3

File: concat-css.php [hunk 4]
```diff
@@ -112,11 +128,11 @@ function do_items( $handles = false, $group = false ) {
 					$media = 'all';
 				}
 
-				if ( ! isset( $stylesheets[ $stylesheet_group_index ] ) || ( isset( $stylesheets[ $stylesheet_group_index ] ) && ! is_array( $stylesheets[ $stylesheet_group_index ] ) ) ) {
-					$stylesheets[ $stylesheet_group_index ] = array();
+				if ( ! isset( $stylesheets[ $concat_group ] ) || ( isset( $stylesheets[ $concat_group ] ) && ! is_array( $stylesheets[ $concat_group ] ) ) ) {
+					$stylesheets[ $concat_group ] = array();
```
Comment (brandonpayton) [inline concat-css.php:132]: Why is this necessary? We explicitly initialize `$stylesheets[ $concat_group ]` [here](https://github.com/Automattic/page-optimize/pull/10/files?w=1#diff-1ffea94758f96baec2cef05252ccf839R41).

**Review by brandonpayton — APPROVED** [review]
This looks good to me. Nice work!

I left one comment about code that looked unnecessary, but it can be removed later if it makes sense.

File: settings.php (hunk 2 above)
Comment (aidvu) [reply settings.php:75]: Very true. :)

File: concat-css.php (hunk 4 above)
Comment (aidvu) [reply concat-css.php:132]: Yeah, left it from going back and forth with testing. Totally not needed.
